psse/psse-model/src/main/java/com/powsybl/psse/model/PsseRawReader.java

```diff
@@ -21,39 +23,255 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
  */
 public class PsseRawReader {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(PsseRawReader.class);
 
-    private static String removeComment(String line) {
-        int slashIndex = line.lastIndexOf('/');
-        if (slashIndex == -1) {
-            return line;
+    public boolean checkCaseIdentification(BufferedReader reader) throws IOException {
+        Objects.requireNonNull(reader);
+
+        // just check the first record if this file is in PSS/E format
+        PsseCaseIdentification caseIdentification;
+        try {
+            caseIdentification = readCaseIdentificationData(reader);
+        } catch (PsseException e) {
+            return false; // invalid PSS/E content
         }
-        return line.substring(0, slashIndex);
+
+        int ic = caseIdentification.getIc();
+        double sbase = caseIdentification.getSbase();
+        int rev = caseIdentification.getRev();
+        double basfrq = caseIdentification.getBasfrq();
+
+        if (ic == 0 && sbase > 0. && rev <= PsseConstants.SUPPORTED_VERSION && basfrq > 0.) {
+            return true;
+        }
+
+        return false;
```

Comment:
I think we should at least log which assertion is not true, specially if the version is not supported.

Reply:
Done